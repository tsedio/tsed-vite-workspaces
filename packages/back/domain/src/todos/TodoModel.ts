import { Model, ObjectID } from "@tsed/mongoose";
import { CollectionOf, Default, Enum, Example, Groups, MaxItems, MaxLength, MinLength, Required } from "@tsed/schema";

import { TaskModel } from "./TaskModel";
import { TodoStatuses } from "./TodoStatuses";

@Model({ name: "todos" })
export class TodoModel {
  @ObjectID("id")
  @Groups("!creation")
  _id: string;

  @Required()
  @MinLength(3)
  @MaxLength(200)
  @Example("My todo title")
  title: string;

  @CollectionOf(TaskModel)
  @Required()
  @MaxItems(20)
  tasks: TaskModel[] = [];

  @Enum(TodoStatuses)
  @Required()
  @Default(TodoStatuses.CREATED)
  status: TodoStatuses = TodoStatuses.CREATED;
}
