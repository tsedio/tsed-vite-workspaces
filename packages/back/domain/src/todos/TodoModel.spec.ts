import { PlatformTest } from "@tsed/common";
import { getJsonSchema } from "@tsed/schema";

import { TodoModel } from "./TodoModel";

describe("TodoModel", () => {
  beforeEach(PlatformTest.create);
  afterEach(PlatformTest.reset);

  it("generate the expected schema", () => {
    const schema = getJsonSchema(TodoModel);
    expect(schema).toMatchSnapshot();
  });
});
