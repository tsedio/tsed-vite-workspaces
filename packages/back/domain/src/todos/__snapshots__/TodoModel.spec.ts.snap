// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoModel generate the expected schema 1`] = `
Object {
  "definitions": Object {
    "TaskModel": Object {
      "properties": Object {
        "done": Object {
          "default": false,
          "type": "boolean",
        },
        "title": Object {
          "examples": Array [
            "My task title",
          ],
          "maxLength": 200,
          "minLength": 3,
          "type": "string",
        },
      },
      "required": Array [
        "title",
        "done",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "id": Object {
      "description": "An ObjectID",
      "examples": Array [
        "5ce7ad3028890bd71749d477",
      ],
      "pattern": "^[0-9a-fA-F]{24}$",
      "type": "string",
    },
    "status": Object {
      "default": "CREATED",
      "enum": Array [
        "CREATED",
        "IN_PROGRESS",
        "DONE",
      ],
      "minLength": 1,
      "type": "string",
    },
    "tasks": Object {
      "items": Object {
        "$ref": "#/definitions/TaskModel",
      },
      "maxItems": 20,
      "type": "array",
    },
    "title": Object {
      "examples": Array [
        "My todo title",
      ],
      "maxLength": 200,
      "minLength": 3,
      "type": "string",
    },
  },
  "required": Array [
    "title",
    "tasks",
    "status",
  ],
  "type": "object",
}
`;
